glue/plugins/wcs_autolinking/tests/test_wcs_autolinking.py

```diff
@@ -240,3 +241,254 @@ def test_celestial_with_unknown_axes():
+    """
+
+    wcs1 = WCS(naxis=4)
+    wcs1.wcs.ctype = 'WAVE', 'HPLT-TAN', 'HPLN-TAN', 'TIME'
+    wcs1.wcs.set()
+
+    data1 = Data(label='Data 1')
+    data1.coords = wcs1
+    data1['x'] = np.ones((2, 3, 4, 5))
+    pw1, pz1, py1, px1 = data1.pixel_component_ids
+
+    wcs2 = WCS(naxis=3)
+    wcs2.wcs.ctype = 'HPLN-TAN', 'HPLT-TAN', 'TIME'
+    wcs2.wcs.set()
+
+    data2 = Data(label='Data 2')
+    data2.coords = wcs2
+    data2['x'] = np.ones((2, 3, 4))
+    pz2, py2, px2 = data2.pixel_component_ids
+
+    dc = DataCollection([data1, data2])
+    links = wcs_autolink(dc)
+    assert len(links) == 1
+    link = links[0]
+    assert isinstance(link, MultiLink)
+    assert len(link) == 6
+    assert link[0].get_to_id() == px2
+    assert link[0].get_from_ids() == [py1, pz1, pw1]
+    assert link[1].get_to_id() == py2
+    assert link[1].get_from_ids() == [py1, pz1, pw1]
+    assert link[2].get_to_id() == pz2
+    assert link[2].get_from_ids() == [py1, pz1, pw1]
+    assert link[3].get_to_id() == py1
+    assert link[3].get_from_ids() == [px2, py2, pz2]
+    assert link[4].get_to_id() == pz1
+    assert link[4].get_from_ids() == [px2, py2, pz2]
+    assert link[5].get_to_id() == pw1
+    assert link[5].get_from_ids() == [px2, py2, pz2]
+
+
+def test_2d_and_1d_data_cubes_with_no_celestial_axes():
+    """
+    Test the case where we have one 2D dataset with WAVE and TIME 
+    as CTYPEs and a 1D dataset with WAVE as the CTYPE
+    """
+
+    wcs1 = WCS(naxis=2)
+    wcs1.wcs.ctype = 'TIME', 'WAVE'
+    wcs1.wcs.set()
+
+    data1 = Data(label='Data 1')
+    data1.coords = wcs1
+    data1['x'] = np.ones((2, 3))
+    py1, px1 = data1.pixel_component_ids
+
+    wcs2 = WCS(naxis=1)
+    wcs2.wcs.ctype = ['WAVE']
+    wcs2.wcs.set()
+
+    data2 = Data(label='Data 2')
+    data2.coords = wcs2
+    data2['x'] = np.ones(3)
+    px2 = data2.pixel_component_ids
+
+    dc = DataCollection([data1, data2])
+    links = wcs_autolink(dc)
+    assert len(links) == 1
+    link = links[0]
+    assert isinstance(link, MultiLink)
+    assert len(link) == 2
+    assert ' '.join(str(link[0].get_to_id()).split()[:2]) == ' '.join(str(py1).split()[:2])
+    assert ' '.join(str(link[0].get_from_ids()).split()[:2]) == ' '.join(str(px2).split()[:2])
+
+
+def test_link_of_spectral_axes_of_different_physical_types():
+    """
+    To check that there is no auto-link of spectral axes of two different physical types, e.g.
+    between FREQ and WAVE.
+    TODO: To modify the codebase so that there will be a link between FREQ and WAVE.
+    """
+
+    wcs1 = WCS(naxis=1)
+    wcs1.wcs.ctype = ['FREQ']
+    wcs1.wcs.set()
+
+    data1 = Data(label='Data 1')
+    data1.coords = wcs1
+    data1['x'] = np.ones(2)
+
+    wcs2 = WCS(naxis=1)
+    wcs2.wcs.ctype = ['WAVE']
+    wcs2.wcs.set()
+
+    data2 = Data(label='Data 2')
+    data2.coords = wcs2
+    data2['x'] = np.ones(2)
+
+    dc = DataCollection([data1, data2])
+    links = wcs_autolink(dc)
+    assert len(links) == 0
```

Comment:
Can you write the test as if it was supposed to work, remove the TODO, and add an xfail decorator? Then it makes it easier once we do actually implement support for this.

Reply:
It is interesting, but I just ran the test again and now it seems there is indeed a link between the `FREQ` and `WAVE`... Not sure what changed, except I switched my `astropy` version to `4.1rc1`.

Reply:
This could be a bug, however...